### ISSUE TITLE
RT/Fav/Follow/... Notifications

### ISSUE THREAD

**Issue by baedert** [baedert/corebird#317]
- No support for retweet notifications on the streaming connection. Options:
    - Support it only for followers, since whenever they retweet a tweet of the user, we get it as a normal tweet (and then don't show it).
    - Polling.

- No support for favorite notifications on the streaming connection. Options:
    - Support it only for followers (since we get a message on the stream there, see RTs).
    - Polling.

- Being followed seems to just work.

**Comment by seboss666** [comment]
I agree with that. As it is standing for now, Corebird is almost on par with the web client. But you only get direct mentions, not retweets/favorites/list addition notifications.

It's not far to entirely ditch the website to use it instead (freeing some CPU&memory in Firefox).

**Comment by baedert** [comment]
Uhm, I just listed some facts -- what is possible and what isn't. What do you agree with?

**Comment by seboss666** [comment]
I misunderstood a little (sorry, I'm french, and even when being comfortable in english, I can be wrong sometimes). Also, I did not read at the Twitter API to see what's possible. That said, I think polling would be too "fat" to implement, so you have to choose between partial notification support at least from followers for RT/fav, or no support at all. If I'm correct. So my personal view would be to implement the partial support :)

**Comment by steinuil** [comment]
The Twitter Mac client only supports rt/fav notifications for people you follow, I think implementing that would be alright.
Anyway, correct me if I'm wrong, but the REST API seems to provide a GET request for retweets of the user's tweets though I'm not sure if it includes RTs by people you don't follow. If that were the case, would it be possible to combine the two and stream RT notifications by followers, and make a request every minute from the REST API for the others?

**Comment by baedert** [comment]
Yes. As I said, polling is an option.

**Comment by baedert** [comment]
Correction:

  - There's no API resource to query the tweets of the user that have been favorited by others.
  - The API endpoint to query the tweets of the user that have been retweeted by others does *not* include who retweeted them. So we'd have to get the tweet ID here, then query all the retweeter's ID(of course, filter them after which we already notified for!), then get their user info, and *then* we can display it, and that's just ridiculous.

Little tired atm, but that seems to be the situation.

**Comment by steinuil** [comment]
Wow, thanks twitter. Yeah, I guess implementing it only for users you follow is the only sane option.

**Comment by ryanleesipes** [comment]
Has this changed?

**Comment by baedert** [comment]
Not that I know of
